### PR TITLE
feat: add smooth L1 loss

### PR DESCRIPTION
**Pull request by DonsetPG**

### CODE REVIEW

**Comment by UgoPelissier** [comment]
In torch.nn.modules, `beta: float = 1.0` is specified in the definition of __init__(), should we do the same?

https://github.com/pytorch/pytorch/blob/v2.6.0/torch/nn/modules/loss.py#L962 (Line 1036)

**Review by jviquerat — APPROVED** [review]

**Review by UgoPelissier — APPROVED** [review]
